bin/runtime/src/lib.rs

```diff
@@ -750,6 +751,84 @@ impl pallet_identity::Config for Runtime {
     type RegistrarOrigin = EnsureRoot<AccountId>;
     type WeightInfo = pallet_identity::weights::SubstrateWeight<Self>;
 }
+parameter_types! {
+    // Key size = 32, value size = 8
+    pub const ProxyDepositBase: Balance = 40 * LEGACY_DEPOSIT_PER_BYTE;
+    // One storage item (32) plus `ProxyType` (1) encode len.
+    pub const ProxyDepositFactor: Balance = 33 * LEGACY_DEPOSIT_PER_BYTE;
+    // Key size = 32, value size 8
+    pub const AnnouncementDepositBase: Balance =  40 * LEGACY_DEPOSIT_PER_BYTE;
+    // AccountId, Hash and BlockNumber sum up to 68
+    pub const AnnouncementDepositFactor: Balance =  68 * LEGACY_DEPOSIT_PER_BYTE;
+}
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    RuntimeDebug,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    Any = 0,
+    NonTransfer = 1,
+    Staking = 2,
+}
+impl Default for ProxyType {
+    fn default() -> Self {
+        Self::Any
+    }
+}
+impl InstanceFilter<RuntimeCall> for ProxyType {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match self {
+            ProxyType::Any => true,
+            ProxyType::NonTransfer => !matches!(
+                c,
+                RuntimeCall::Balances(..)
+                    | RuntimeCall::Vesting(pallet_vesting::Call::vested_transfer { .. })
+            ),
+            ProxyType::Staking => {
+                matches!(
+                    c,
+                    RuntimeCall::Staking(..)
+                        | RuntimeCall::Session(..)
+                        | RuntimeCall::Utility(..)
+                        | RuntimeCall::NominationPools(..)
+                )
+            }
+        }
+    }
+    fn is_superset(&self, o: &Self) -> bool {
```

Comment:
What is this responsible for?

Reply:
In pallet proxy its used in one place.
```
// Proxy call cannot add or remove a proxy with more permissions than it already has.
```
So proxy calls that add or remove proxy (A) of some type have to be called with a proxy (B) such that `B.is_superset(A) == true`

Reply:
I propose to write this like:

```
    fn is_superset(&self, o: &Self) -> bool {
        match (self, o) {
            (x, y) if x == y => true,
            (ProxyType::Any, ProxyType::NonTransfer) => true,
            (ProxyType::Any, ProxyType::Staking) => true,
            (ProxyType::NonTransfer, ProxyType::Staking) => true,
            _ => false,
        }
    }
```

This way it's not possible to make any mistake when adding a new proxy type in the future.

Reply:
Ok, I have rewritten it but in a little bit different way. Removed capture all arm and `(x, y) if x == y => true`.
First 2 arms to comfirm to `Any` specification
```
            (ProxyType::Any, _) => true,
            (_, ProxyType::Any) => false,
```
And the rest is explicitly written.